SimpleChat.Server/Controllers/UsersController.cs

```diff
@@ -0,0 +1,98 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Net.Mime;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using SimpleChat.Bll.Extensions;
+using SimpleChat.Bll.Interfaces;
+using SimpleChat.Server.Extensions;
+using SimpleChat.Shared.Contracts;
+
+namespace SimpleChat.Server.Controllers
+{
+    /// <summary>
+    /// Users controller.
+    /// </summary>
+    [Produces(MediaTypeNames.Application.Json)]
+    [Route("api/[controller]")]
+    [ApiController]
+    public class UsersController : ControllerBase
+    {
+        private readonly IUserService service;
+
+        /// <summary>
+        /// Constuctor.
+        /// </summary>
+        /// <param name="service">Injects <see cref="IUserService"/> in controller.</param>
+        public UsersController(IUserService service)
+        {
+            this.service = service;
+        }
+
+        /// <summary>
+        /// Gets user by id.
+        /// </summary>
+        /// <param name="id">User id.</param>
+        /// <returns>Instance of <see cref="User"/>.</returns>
+        /// <response code="200">Returns user.</response>           
+        [HttpGet("{id}")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        public async Task<User> GetAsync(int id)
+        {
+            var userModel = await service.GetByIdAsync(id);
+            var userContract = userModel.ToContract();
+            //if user == httpcontext.Session.User
+            userContract.Chats = userModel.Chats.Select(x => x.ToContract());
+            
+            return userContract;
+        }
+
+        /// <summary>
+        /// Gets users by Chat id.
+        /// </summary>
+        /// <param name="id">Chat id.</param>
+        /// <param name="pagination">Presents data for pagination.</param>
+        /// <returns><see cref="IEnumerable{T}"/>Collection of users.</returns>
+        /// <response code="200">Returns user.</response>           
+        [HttpGet("[action]")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        public async Task<IEnumerable<User>> GetByChatAsync(int id, [FromQuery] Pagination pagination) =>
```

Comment:
Do you really need this action?

Reply:
I think it's important for getting a list of chat users